2_process/src/munge_GFv1_catchments.R

```diff
@@ -0,0 +1,147 @@
+  #' 
+  #' @description This function takes a data table of PRMS segment ID's and
+  #' its contributing NHDv2 tributary catchments and returns those catchments
+  #' as an sf object
+  #' 
+  #' @param prms_line sf object containing one PRMS river segment of interest
+  #' prms_line must contain variables subsegid and geometry
+  #' @param segs_w_comids data frame containing the PRMS segment ids and the contributing COMID's
+  #' segs_w_comids must contain variables PRMS_segid and COMID
+  #' 
+  
+  # Subset segs_w_comids to only include the PRMS segment of interest
+  nhd_cats <- segs_w_comids %>%
+    filter(PRMS_segid == prms_line$subsegid)
+  
+  # Use nhdplusTools to retrieve the catchment polygons for COMIDs of interest
+  nhd_cats_sf <- nhdplusTools::get_nhdplus(comid = c(nhd_cats$COMID),realization = "catchment",t_srs = 5070) %>%
+    # dissolve boundaries between individual NHDv2 catchments
+    sf::st_union() %>%
+    # convert back to sf data.frame object and add the segment ID
+    sf::st_as_sf() %>%
+    mutate(PRMS_segid = unique(prms_line$subsegid)) %>%
+    rename(geometry = x)
+  
+  return(nhd_cats_sf)
+  
+}
+
+
+
+munge_GFv1_catchments <- function(prms_lines,prms_hrus,segs_w_comids,crs_out = 4326){
+  #' 
+  #' @description This function munges the PRMS HRU's (~ the PRMS catchments) so that every
+  #' PRMS segment in the Delaware River Basin has at least one corresponding HRU and so that 
+  #' the catchment boundaries are adjusted for 3 NHGFv1 segments that were split in the 
+  #' `delaware-model-prep` pipeline: segid's 3_1, 3_2, 8_1, 8_2, 51_1, 51_2 
+  #' (https://github.com/USGS-R/delaware-model-prep) 
+  #' 
+  #' @param prms_lines sf object containing the PRMS river segments for the DRB
+  #' prms_lines must contain variables subsegid and geometry
+  #' @param segs_w_comids data frame containing the PRMS segment ids and the contributing COMID's
+  #' segs_w_comids must contain variables PRMS_segid and COMID
+  #' @param prms_hrus sf (multi)polygon containing the HRU's from the NHGFv1
+  #' @param crs_out numeric EPSG code indicating desired crs. Defaults to EPSG: 4326, WGS84.
+  #
+  
+  # 1. Fill PRMS lines data frame so that every segment has a value for the attribute
+  # `hru_segment`, which can be used as a key to join each segment to its corresponding HRU polygons.
+  prms_hrus_filled <- prms_lines %>%
+    # Transform prms_lines into a list with a data frame for each PRMS segment (represented by a unique subsegid) 
+    split(.,.$subsegid) %>%
+    # For each PRMS segment, find the HRU with the greatest overlap and add as an attribute
+    lapply(.,function(x){
+      x$hru_segment <- find_intersecting_hru(x,prms_hrus)
+      return(x)
+    }) %>%
+    # Reformat list into a data frame with one row per PRMS segment
+    bind_rows()
+  
+  # 2. For select segments, redefine the catchment polygon using the contributing NHDv2 catchments
+  segs_split <- c("3_1","3_2","8_1","8_2","51_1","51_2")
```

Comment:
I don't expect this character vector to change (at least for a network based on NHGFv1), but I like this suggestion to move `segs_split` to the `_targets.R` file because the segment values are now more visible.